pkg/endpoint/bpf.go

```diff
@@ -144,6 +144,18 @@ func (e *Endpoint) writeHeaderfile(prefix string) error {
 	}
 	defer f.Cleanup()
 
+	// Update DNSRules if any. This is needed because DNSRules also encode allowed destination IPs
+	// and those can change anytime we have identity updates in the cluster. If there are no
+	// DNSRules (== nil) we don't need to update here, as in that case there are no allowed
+	// destinations either.
+	if e.DNSRules != nil {
+		e.OnDNSPolicyUpdateLocked(e.owner.GetDNSRules(e.ID))
```

Comment:
I thought we were deliberately only updating this when the policy changes to reduce load here :-)

This runs every 5s. Are we happy enough with the performance in pathological cases to do this frequently?

Reply:
The set of selected IPs can change any time. The cost of figuring out if any of the selectors now selects new IDs and whether the IPs they map to are still the same is roughly the same as recreating the structure anyway?

Reply:
Maybe we should have some limit for the max number of rules and/or IPs we process?

Reply:
> The set of selected IPs can change any time

Ah right, I keep thinking of this as storing the policy layer but we must realize the selectors to provide meaningful data on restore. :+1: 

> The cost of figuring out if any of the selectors now selects new IDs and whether the IPs they map to are still the same is roughly the same as recreating the structure anyway?

:+1: 

> Maybe we should have some limit for the max number of rules and/or IPs we process?

That seems somewhat reasonable if we're trying to bound the time, but without knowing the cost it's hard to know what sensible values would be here or whether we would simply introduce additional problems by adding a limiter. 

Are there simple high-signal outputs we could get here that would point when if this is *actually* expensive at runtime, not just based on my wild speculation? I'm thinking again something like metrics or a very rare warning message. Seems like we expect this to be pretty cheap despite the `GetRules()` nesting that Chris pointed out in the thread above (https://github.com/cilium/cilium/pull/12718#discussion_r463154010).

Reply:
Resolved in the latest patch.